jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/deadlines/notifications/impl/email/EmailNotificationListener.java

```diff
@@ -114,11 +114,12 @@ public void onNotification(NotificationEvent event, UserInfo userInfo) {
                     for (User user : entry.getValue()) {
     
                         String emailAddress = userInfo.getEmailForEntity(user);
-                        if (emailAddress != null && !toAddresses.contains(emailAddress)) {                        	
-                        	msg.addRecipients( Message.RecipientType.TO, InternetAddress.parse( emailAddress, false));
-                        	toAddresses.add(emailAddress);
+                        if (emailAddress != null) {                        	
+                        	if (toAddresses.add(emailAddress)) {
```

Comment:
I think this can be combined into ```if (emailAddress != null && toAddresses.add(emailAddress))```

Reply:
that was the whole point to split it to avoid warning saying that email was not found in case the set already has the email address

Reply:
Sorry, I didn't notice the else part :)